### PR TITLE
squashfuse 0.1.103 (new formula)

### PR DESCRIPTION
**Pull request by get9**
New formula for squashfuse.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by commitay** [comment]
No it isn't a bug in audit. 

I suggest requiring `squashfs` and the two compression algorithms it requires `lzo` and `xz`.

**Comment by get9** [comment]
You can also optionally use zstd as a compression algorithm - should that be included as well? What's the motivation for removing options on new formulae?

**Comment by get9** [comment]
@commitay: Is this more like what you were looking for?

**Comment by get9** [comment]
@commitay @fxcoudert: Do I need to do anything else for this formula or is it good to go?

File: Formula/squashfuse.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+class Squashfuse < Formula
+  desc "FUSE filesystem to mount squashfs archives"
+  homepage "https://github.com/vasi/squashfuse"
+  url "https://github.com/vasi/squashfuse/archive/0.1.103.tar.gz"
```
Comment (ilovezfs) [inline Formula/squashfuse.rb]: We should be using https://github.com/vasi/squashfuse/releases/download/0.1.103/squashfuse-0.1.103.tar.gz and avoiding the Autotools dependencies.

File: Formula/squashfuse.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  def install
+    configure_args = [
+      "--disable-dependency-tracking",
+      "--disable-silent-rules",
+      "--prefix=#{prefix}",
+    ]
+
+    %w[lz4 lz0 xz zlib zstd].each do |algorithm|
+      configure_args << "--with-#{algorithm}=#{HOMEBREW_PREFIX}"
```
Comment (ilovezfs) [inline Formula/squashfuse.rb]: should use opt prefixes like `Formula["lz4"].opt_prefix` but I doubt the explicit specifications are needed anyway. does it not find the deps on its own?

File: Formula/squashfuse.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    end
+    system "./autogen.sh"
+    system "./configure", *configure_args
+    system "make", "install"
+  end
+
+  # Unfortunately, making/testing a squash mount requires sudo priviledges, so
+  # just test that squashfuse execs for now.
+  test do
+    pid = fork { exec "squashfuse", "--help" }
```
Comment (ilovezfs) [inline Formula/squashfuse.rb]: should be in a `begin` block with Process.kill in `ensure`

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/squashfuse.rb (hunk 2 above)
Comment (get9) [reply Formula/squashfuse.rb]: Yeah, just tested it and it was able to find homebrew-installed libs. Will remove this.

**Comment by get9** [comment]
@ilovezfs More like this?

**Comment by get9** [comment]
@ilovezfs: ping?

File: Formula/squashfuse.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
+class Squashfuse < Formula
+  desc "FUSE filesystem to mount squashfs archives"
+  homepage "https://github.com/vasi/squashfuse"
+  url "https://github.com/vasi/squashfuse/releases/download/0.1.103/squashfuse-0.1.103.tar.gz"
+  sha256 "42d4dfd17ed186745117cfd427023eb81effff3832bab09067823492b6b982e7"
+
+  depends_on "pkg-config" => :build
+
```
Comment (DomT4) [inline Formula/squashfuse.rb]: Don't need the empty line here.

File: Formula/squashfuse.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  depends_on "lz4"
+  depends_on "lzo"
+  depends_on :osxfuse
+  depends_on "squashfs"
+  depends_on "xz"
+  depends_on "zstd"
+
+  def install
+    configure_args = [
```
Comment (DomT4) [inline Formula/squashfuse.rb]: Since you're not passing any extra `args` just inline these.

File: Formula/squashfuse.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  test do
+    def pid_exists?(pid)
+      Process.kill 0, pid
+      true
+    rescue Errno::ESRCH
+      false
+    end
+
+    begin
+      pid = fork { exec "squashfuse", "--help" }
```
Comment (DomT4) [inline Formula/squashfuse.rb]: `bin/"squashfuse"`

**Review by ilovezfs — APPROVED** [review]

**Pull request closed without merging**